openjdk-integ-tests/src/test/java/libcore/javax/net/ssl/SSLSocketTest.java

```diff
@@ -1647,9 +1655,13 @@ private void test_SSLSocket_interrupt_case(boolean readUnderlying, boolean close
         try {
             toRead.setSoTimeout(readingTimeoutMillis);
             final InputStream inputStream = toRead.getInputStream();
-            @SuppressWarnings("unused")
-            int value = inputStream.read();
-            fail();
+            @SuppressWarnings("unused") int value = inputStream.read();
+            // This seems to only happen with Conscrypt's FD-based socket.
+            assertTrue(readUnderlying);
+            assertFalse(closeUnderlying);
+            assertEquals("OpenSSLSocketImplWrapper", clientWrapping.getClass().getSimpleName());
+            assertEquals(-1, value);
+            throw new SocketException();
```

Comment:
This shouldn't throw here. It should assert the opposite condition in the `catch` block and then change the name from `expected` to `maybe` or something like that.